src/python/pants/backend/adhoc/code_quality_tool.py

```diff
@@ -366,20 +365,19 @@ async def partition_inputs(
         async def run_code_quality(request: CodeQualityProcessingRequest.Batch) -> FmtResult:
             sources_snapshot = request.snapshot
 
-            code_quality_tool_runner = await Get(
-                ToolRunner, CodeQualityToolAddressString(address=self.target)
-            )
+            cqt = await find_code_quality_tool(CodeQualityToolAddressString(address=self.target))
+            code_quality_tool_runner_request = await runner_request_for_code_quality_tool(cqt)
+            code_quality_tool_runner = await create_tool_runner(code_quality_tool_runner_request)
```

Comment:
Alternatively, this could use implicitly like the `create_tool_runner` call site [above](https://github.com/pantsbuild/pants/pull/22354/files#diff-6d468a45ffadf6d61cf3cb333323d77e684e0274b3c3f1a96f10d751c2081c6aR316):
```suggestion
            code_quality_tool_runner = await create_tool_runner(
                **implicitly(CodeQualityToolAddressString(address=self.target))
            )
```
And we could use implicitly at one more call site [below](https://github.com/pantsbuild/pants/pull/22354/files#diff-6d468a45ffadf6d61cf3cb333323d77e684e0274b3c3f1a96f10d751c2081c6aR431).

I expanded it because it felt a little weird to use implicitly to refer to 2 rules that are in this file: `find_code_quality_tool` and `runner_request_for_code_quality_tool`.

Any preferences here?